src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/jpa/BuildingRepository.kt

```diff
@@ -19,27 +21,31 @@ import javax.persistence.OneToMany
 interface BuildingRepository : CrudRepository<Building, Long>
 
 @Entity
+@EntityListeners(AuditingEntityListener::class)
 data class Building(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   val id: Long? = null,
+
   @ManyToOne(optional = false, fetch = FetchType.LAZY)
   @JoinColumn(name = "COURT_CODE")
-  val court: Court,
-  val subCode: String?,
-  val buildingName: String?,
-  val street: String?,
-  val locality: String?,
-  val town: String?,
-  val county: String?,
-  val postcode: String?,
-  val country: String?,
+  var court: Court,
```

Comment:
It's a shame about these `var`s... this article seems to avoid them (apologies if you've already tried this!) https://www.baeldung.com/kotlin/jpa#compiler-plugins-jpa-plugin